consensus/blocknode.go

```diff
@@ -58,143 +41,103 @@ type blockNode struct {
 	delayedSiacoinOutputs map[SiacoinOutputID]SiacoinOutput
 }
 
-// CheckTarget returns true if the block id is lower than the target.
-func (b Block) CheckTarget(target Target) bool {
-	blockHash := b.ID()
-	return bytes.Compare(target[:], blockHash[:]) >= 0
+// childDepth returns the depth of a blockNode's child nodes. The depth is the
+// "sum" of the current depth and current target, where sum is defined as:
+//
+//     sum(x,y) := 1/(1/x + 1/y)
+func (bn *blockNode) childDepth() (depth Target) {
+	cumulativeDifficulty := new(big.Rat).Add(bn.target.Inverse(), bn.depth.Inverse())
+	return RatToTarget(new(big.Rat).Inv(cumulativeDifficulty))
 }
 
-// Int returns a Target as a big.Int.
-func (t Target) Int() *big.Int {
-	return new(big.Int).SetBytes(t[:])
+// heavierThan returns true if the blockNode is sufficiently heavier than
+// 'cmp', where "sufficient" is defined as:
+//
+//     (1/bn.depth) - (1/cmp.depth) > (1/cmp.target) * SurpassThreshold
+func (bn *blockNode) heavierThan(cmp *blockNode) bool {
```

Comment:
I'm not sure I understand the change here. Why are we now comparing two blocknodes instead of comparing the blocknode to the state? I get that it's more general, but when you are commencing a fork you definitely need to be comparing against the state.

It seems like a potential place to make mistakes. But I guess there could be some value in knowing if block 500 on fork A is heavier than block 500 on fork B, though I don't think that in practice we'll ever see forks longer than 3 blocks, and probably not something you'd explore from within the core application.